### ISSUE TITLE
Create Article API ignores tags param

### ISSUE THREAD

**Issue by worsnupd** [thepracticaldev/dev.to#4497]
**Describe the bug**

When I post a new article to https://dev.to/api/articles, the `tags` param is ignored.

**Comment by rhymes** [comment]
@worsnupd thanks for this report!

can you be more specific or post a "curl" command that can be used to reproduce the bug?

**Comment by 3v0k4** [comment]
Today I got stuck on this too. Then I realized that I was not sending the params inside the `"article"` root key. In other words, I was sending

```
{
  "title": "some title",
  "tags": ["tag1", "tag2"]
}
```

instead of 

```
{
  "article": {
    "title": "some title",
    "tags": ["tag1", "tag2"]
  }
}
```

This is confusing because the first request works but it "ignores" `tags`. That's because of `wrap_parameters` and how params are accessed in the controller `params.permit(:article)...`.

From the [`ParamsWrapper` docs](https://api.rubyonrails.org/v6.0.1/classes/ActionController/ParamsWrapper.html):

> The ParamsWrapper will actually try to determine the list of attribute names from the model and only wrap those attributes

For that reason, `title` is available in the controller regardless of the presence of the `"article"` root key in the JSON. The same is true for all `Article.attribute_names`.  However, `tags` is not part of `Article`'s attribute. Thus, it works only when it's correctly nested.

I guess there could be some validation logic returning an error instead of succeeding with part of the parameters. But in the end, my bad for not reading the docs well 👍
